mitodl/mit-open
# Fix api urls

# What are the relevant tickets?
closes https://github.com/mitodl/mit-open/issues/280
closes https://github.com/mitodl/mit-open/issues/283

# Description (What does it do?)
This pr standardizes the names of path variables in the documentation.
Additionally it fixes the issue that nested routes such as api/v1/learning_resources/[learning_resource]/items/[id] required the LearningResourceRelationship id that was not obtainable from the api. Now api/v1/learning_resources/[learning_resource]/items/ returns information about both the nested resources and the relationship

# How can this be tested?
1) Test that you can create and edit learning paths at http://localhost:8063/learningpaths/
and add items from the front page carousel to a learning path
2) Go to http://localhost:8063/api/v1/schema/swagger-ui/ and check that the names of the path variables in the urls are standardized  to `learning_resource_id` or `userlist_id` instead of the original names like `run__learning_resource` and `parent_id`. The path variables should also have a description explaining what they are
3) Test out the nested api calls under api/v1/learning_resources and api/v1/learningpaths and api/v1/userlists. They should work
4) Additionally the results for api/v1/learning_resources/[learning_resource]/items/ should show both the resources and the relationship information.
